userspace/otpilot/src/spi_processor.rs

```diff
@@ -159,30 +169,191 @@ impl<'a> SpiProcessor<'a> {
     }
 
     fn process_manticore(&mut self, mut data: &[u8]) -> SpiProcessorResult<()> {
-        let mut console = Console::new();
-        writeln!(console, "Device: Manticore!")?;
+        let payload_len : u16;
+        {
+            unsafe {
+                // TODO(osk): We need the unsafe block since we're accessing SPI_TX_BUF as &mut.
+                let mut tx_cursor = ManticoreCursor::new(&mut SPI_TX_BUF[payload::HEADER_LEN..]);
+                self.server.process_request(&mut data, &mut tx_cursor)?;
+                payload_len = u16::try_from(tx_cursor.consumed_len())
+                    .map_err(|_| SpiProcessorError::FromWire(FromWireError::OutOfRange))?;
+            }
+        }
+        unsafe {
+            // TODO(osk): We need the unsafe block since we're accessing SPI_TX_BUF as &mut.
+            self.send_data(payload::ContentType::Manticore, payload_len, &mut SPI_TX_BUF)?;
+        }
+        Ok(())
+    }
 
+    fn send_firmware_response<'m, M: Message<'m>>(&mut self, response: M) -> SpiProcessorResult<()> {
         let payload_len : u16;
         unsafe {
             // TODO(osk): We need the unsafe block since we're accessing SPI_TX_BUF as &mut.
-            let mut tx_cursor = ManticoreCursor::new(&mut SPI_TX_BUF[payload::HEADER_LEN..]);
-            self.server.process_request(&mut data, &mut tx_cursor)?;
+            let mut tx_cursor = SpiutilsCursor::new(&mut SPI_TX_BUF[payload::HEADER_LEN..]);
+
+            let fw_header = firmware::Header {
+                content: M::TYPE
+            };
+            fw_header.to_wire(&mut tx_cursor)?;
+            response.to_wire(&mut tx_cursor)?;
             payload_len = u16::try_from(tx_cursor.consumed_len())
-                .map_err(|_| SpiProcessorError::FromWire(FromWireError::OutOfRange))?;
+                    .map_err(|_| SpiProcessorError::FromWire(FromWireError::OutOfRange))?;
```

Comment:
Nit: weird indentation.

Reply:
Oh yeah, how did that happen? Thanks!

I'll need to run `rustfmt` on `spiutils`, `otpilot` and the new kernel code that I've written to make sure it conforms. I should have done that before and we should do that going forward. I'll send a follow up PR with just formatting changes.